core/templates/pages/exploration-player-page/services/prediction-algorithm-registry.service.ts

```diff
@@ -43,12 +44,18 @@ export class PredictionAlgorithmRegistryService {
   }
 
   getPredictionService(
-      algorithmId: string, dataSchemaVersion: number): PredictionService {
+      algorithmId: string, dataSchemaVersion: number):
+       PredictionService | null {
     if (this.algorithmIdPredictionServiceMapping.has(algorithmId)) {
       const predictionServicesByDataSchemaVersion = (
         this.algorithmIdPredictionServiceMapping.get(algorithmId));
-      if (predictionServicesByDataSchemaVersion.has(dataSchemaVersion)) {
-        return predictionServicesByDataSchemaVersion.get(dataSchemaVersion);
+      if (predictionServicesByDataSchemaVersion &&
+        predictionServicesByDataSchemaVersion.has(dataSchemaVersion)) {
+        let predictionService =
+         predictionServicesByDataSchemaVersion.get(dataSchemaVersion);
+        if (predictionService) {
```

Comment:
`has()` already checks the condition for `predictionService` to exist but the compiler does not realize that. You are checking two same conditions here. I think a better approach would be to remove the call to `has()` and check the condition for `predictionService` to exist directly.

Reply:
done